pages/api_integration-full-node-streaming.md

```diff
@@ -0,0 +1,439 @@
+				l.Bids[subticks] = append(
+					l.Bids[subticks][:i],
+					l.Bids[subticks][i+1:]...,
+				)
+				break
+			}
+		}
+		if len(l.Bids[subticks]) == 0 {
+			delete(l.Bids, subticks)
+		}
+	} else {
+		for i, o := range l.Asks[subticks] {
+			if o.OrderId == order.OrderId {
+				l.Asks[subticks] = append(
+					l.Asks[subticks][:i],
+					l.Asks[subticks][i+1:]...,
+				)
+				break
+			}
+		}
+		if len(l.Asks[subticks]) == 0 {
+			delete(l.Asks, subticks)
+		}
+	}
+	delete(l.OrderIdToOrder, orderId)
+}
+```
+
+</details>
+
+### StreamOrderbookFill/ClobMatch
+This message is only used to update fill amounts, it does not add or remove orders from the book but can change the quantity remaining for open orders.
+
+The `ClobMatch` data structure contains either a `MatchOrders` or a `MatchPerpetualLiquidation` object. Match Deleveraging events are not emitted. Within each Match object, a `MakerFill` array contains the various maker orders that matched with the singular taker order and the amount of quantums matched.
+
+Note that all matches occur at the maker order price. The `orders` field in the `StreamOrderbookFill` object allow for price lookups based on order id. It contains all the maker order ids, and in the case of non-liquidation orders, it has the taker order.
+
+Mapping each order in `orders` to the corresponding value in the `fill_amounts` field provides the absolute filled amount of quantums that each order is filled to after the ClobMatch was processed. 
+
+
+<details>
+
+<summary>Code Snippet</summary>
+
+```go
+// fillAmountMap is a map of order ids to fill amounts.
+// The SetOrderFillAmount code can be found in `the `OrderUpdateV1` section.
+func (c *GrpcClient) ProcessMatchOrders(
+	matchOrders *clobtypes.MatchOrders,
+	orderMap map[clobtypes.OrderId]clobtypes.Order,
+	fillAmountMap map[clobtypes.OrderId]uint64,
+) {
+	takerOrderId := matchOrders.TakerOrderId
+	clobPairId := takerOrderId.GetClobPairId()
+	localOrderbook := c.Orderbook[clobPairId]
+
+	indexerTakerOrder := v1.OrderIdToIndexerOrderId(takerOrderId)
+	localOrderbook.SetOrderFillAmount(&indexerTakerOrder, fillAmountMap[takerOrderId])
+
+	for _, fill := range matchOrders.Fills {
+		makerOrder := orderMap[fill.MakerOrderId]
+		indexerMakerOrder := v1.OrderIdToIndexerOrderId(makerOrder.OrderId)
+		localOrderbook.SetOrderFillAmount(&indexerMakerOrder, fillAmountMap[makerOrder.OrderId])
+	}
+}
+
+func (c *GrpcClient) ProcessMatchPerpetualLiquidation(
+	perpLiquidation *clobtypes.MatchPerpetualLiquidation,
+	orderMap map[clobtypes.OrderId]clobtypes.Order,
+	fillAmountMap map[clobtypes.OrderId]uint64,
+) {
+	localOrderbook := c.Orderbook[perpLiquidation.ClobPairId]
+	for _, fill := range perpLiquidation.GetFills() {
+		makerOrder := orderMap[fill.MakerOrderId]
+		indexerMakerOrderId := v1.OrderIdToIndexerOrderId(makerOrder.OrderId)
+		localOrderbook.SetOrderFillAmount(&indexerMakerOrderId, fillAmountMap[makerOrder.OrderId])
+	}
+}
+```
+
+</details>
+
+## FAQs
```

Comment:
Because there's another FAQ section within this one with questions and answers, I think this should either be:
- renamed something more general like "Reference Material," or
- deleted and the subsections elevated